### PR TITLE
Construction recipe for barred window with curtains

### PR DESCRIPTION
**Pull request by Kilvoctu**
#### Summary
```SUMMARY: Content "Construction recipe for barred window with curtains"```

#### Purpose of change
I can build a window, a window with curtains, and add bars onto the window. I can't add bars _and_ a curtain? Don't see why it shouldn't be possible.

#### Describe the solution
Extend the current stages of "Install Bars Onto Window" so curtains can be affixed at the end. 

#### Describe alternatives you've considered
Do nothing and continue to have elaborate setups of barred windows and door curtains.

#### Testing
Built a window, installed bars onto the window, then installed curtains onto the barred window. Peeked through the curtains, opened and closed the curtains to test line of sight. Tried walking through the window when it was open and closed. Tore down the curtains.

#### Additional context
Recipe is the same as installing curtains onto a normal window.
Smashing this construction will revert it back to a normal curtainless window with bars.
Graphically will look like barred window when open and curtains when closed.

### CODE REVIEW

File: src/mapdata.cpp [hunk 1]
```diff
@@ -712,8 +712,10 @@ void set_ter_ids()
     t_window_taped = ter_id( "t_window_taped" );
     t_window_domestic = ter_id( "t_window_domestic" );
     t_window_domestic_taped = ter_id( "t_window_domestic_taped" );
+	t_window_bars_domestic = ter_id( "t_window_bars_domestic" );
```
Comment (I-am-Erk) [inline src/mapdata.cpp]:
```suggestion
	  t_window_bars_domestic = ter_id( "t_window_bars_domestic" );
```

File: src/mapdata.cpp [hunk 2]
```diff
@@ -712,8 +712,10 @@ void set_ter_ids()
...
     t_window_open = ter_id( "t_window_open" );
     t_curtains = ter_id( "t_curtains" );
+	t_window_bars_curtains = ter_id( "t_window_bars_curtains" );
```
Comment (I-am-Erk) [inline src/mapdata.cpp]:
```suggestion
	  t_window_bars_curtains = ter_id( "t_window_bars_curtains" );
```

File: data/json/furniture_and_terrain/terrain-windows.json [hunk 3]
```diff
@@ -571,6 +571,65 @@
       "items": [ { "item": "glass_shard", "count": [ 34, 50 ] } ]
     }
   },
+  {
+    "type": "terrain",
+    "id": "t_window_bars_curtains",
+    "name": "window with metal bars and curtains.",
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/terrain-windows.json]:
```suggestion
    "name": "window with metal bars and curtains",
```

File: data/json/furniture_and_terrain/terrain-windows.json [hunk 4]
```diff
@@ -571,6 +571,65 @@
...
+        { "item": "sheet", "count": 2 },
+        { "item": "stick", "count": 1 },
+        { "item": "string_36", "count": 1 }
+      ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_window_bars_domestic",
+    "name": "window with metal bars and curtains.",
```
Comment (Night-Pryanik) [inline data/json/furniture_and_terrain/terrain-windows.json]:
```suggestion
    "name": "window with metal bars and curtains",
```

Comment (Kilvoctu) [reply data/json/furniture_and_terrain/terrain-windows.json]: Commited these changes and astyle test

**Pull request merged**